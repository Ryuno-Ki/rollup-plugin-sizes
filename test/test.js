/* eslint-disable no-console */
"use strict";

const { rollup } = require("rollup");

const plugin = require("../index.js");

describe("rollup-plugin-sizes", () => {
    let spy;

    beforeEach(() => {
        spy = jest.spyOn(global.console, "log");

        spy.mockImplementation(() => { /* NO-OP */ });
    });

    it.each([
        [ "single entry", "./test/specimens/entry.js" ],
        [ "multi-entry array", [
            "./test/specimens/entry.js",
            "./test/specimens/entry2.js",
        ]],
        [ "multi-entry object", {
            entry  : "./test/specimens/entry.js",
            entry2 : "./test/specimens/entry2.js",
        }],
    ])("should describe the sizes of a bundle (%s)", async (name, input) => {
        const bundle = await rollup({
            input,
        
            plugins : [
                plugin({
                    details : true,
                }),
            ],
        });

        await bundle.generate({ format : "es" });

        expect(spy).toHaveBeenCalled();
        expect(spy.mock.calls).toMatchSnapshot();
    });
});

