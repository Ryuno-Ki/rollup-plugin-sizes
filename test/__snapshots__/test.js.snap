// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-sizes should describe the sizes of a bundle (multi-entry array) 1`] = `
Array [
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
  Array [
    "%s:",
    "./test/specimens/entry2.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "346 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths2.js",
    "276 B",
    "79.77",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry2.js",
    "70 B",
    "20.23",
  ],
]
`;

exports[`rollup-plugin-sizes should describe the sizes of a bundle (multi-entry object) 1`] = `
Array [
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
  Array [
    "%s:",
    "./test/specimens/entry2.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "346 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths2.js",
    "276 B",
    "79.77",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry2.js",
    "70 B",
    "20.23",
  ],
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
  Array [
    "%s:",
    "./test/specimens/entry2.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "346 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths2.js",
    "276 B",
    "79.77",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry2.js",
    "70 B",
    "20.23",
  ],
]
`;

exports[`rollup-plugin-sizes should describe the sizes of a bundle (single entry) 1`] = `
Array [
  Array [
    "%s:",
    "./test/specimens/entry.js",
  ],
  Array [
    "%s - %s (%s%%)",
    "app",
    "345 B",
    "100.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "maths.js",
    "276 B",
    "80.00",
  ],
  Array [
    "	%s - %s (%s%%)",
    "entry.js",
    "69 B",
    "20.00",
  ],
]
`;
